engine/src/main/java/org/camunda/bpm/engine/impl/management/SetJobRetriesAsyncBuilderImpl.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.impl.management;
+
+import static org.camunda.bpm.engine.impl.util.EnsureUtil.ensureNotNull;
+
+import java.util.Date;
+import java.util.List;
+
+import org.camunda.bpm.engine.batch.Batch;
+import org.camunda.bpm.engine.history.HistoricProcessInstanceQuery;
+import org.camunda.bpm.engine.impl.ProcessEngineLogger;
+import org.camunda.bpm.engine.impl.SetJobsRetriesByProcessBatchCmd;
+import org.camunda.bpm.engine.impl.cmd.CommandLogger;
+import org.camunda.bpm.engine.impl.cmd.SetJobsRetriesBatchCmd;
+import org.camunda.bpm.engine.impl.interceptor.CommandExecutor;
+import org.camunda.bpm.engine.management.SetJobRetriesAsyncBuilder;
+import org.camunda.bpm.engine.runtime.JobQuery;
+import org.camunda.bpm.engine.runtime.ProcessInstanceQuery;
+
+public class SetJobRetriesAsyncBuilderImpl implements SetJobRetriesAsyncBuilder {
```

Comment:
We can probably compress your Javadocs a little and improve usability of this fluent builder by introducing two more fluent builder classes, `SetJobRetriesByProcessAsyncBuilder` and `SetJobRetriesByJobsAsyncBuilder`, that are returned when calling `#byJobs` or `#byProcess`. These fluent builders either contain job or process selection methods.

API would look something along the lines:

```java
managementService.setJobRetriesAsync(5)
    .byProcess() // returns SetJobRetriesByProcessAsyncBuilder
    .processInstanceIds(...)
    .execute();
```

The following would not work anymore:
```java
managementService.setJobRetriesAsync(5)
    .byProcess() // returns SetJobRetriesByProcessAsyncBuilder
    .jobIds(...)
    .execute();
```